llvm/lib/Target/SystemZ/SystemZISelLowering.cpp

```diff
@@ -8113,10 +8583,151 @@ static bool combineCCMask(SDValue &CCReg, int &CCValid, int &CCMask) {
     CCReg = IPM->getOperand(0);
     return true;
   }
-
   return false;
 }
 
+std::optional<SDValue>
+SystemZTargetLowering::combineSELECT_CC_CCIPMMask(SDNode *N,
+                                                  DAGCombinerInfo &DCI) const {
```

Comment:
I don't understand why we need yet another function here, which is called only for `SELECT_CCMASK` and not `BR_CCMASK`.     Shouldn't all these optimizations apply equally to both cases?  Why cannot this be integrated into `combineCCMask`?